parser.y

```diff
@@ -2457,25 +2457,35 @@ DatabaseOptionList:
  *******************************************************************/
 
 CreateTableStmt:
-	"CREATE" "TABLE" IfNotExists TableName TableElementListOpt CreateTableOptionListOpt PartitionOpt DuplicateOpt AsOpt CreateTableSelectOpt
+	"CREATE" OptTemporary "TABLE" IfNotExists TableName TableElementListOpt CreateTableOptionListOpt PartitionOpt DuplicateOpt AsOpt CreateTableSelectOpt
 	{
-		stmt := $5.(*ast.CreateTableStmt)
-		stmt.Table = $4.(*ast.TableName)
-		stmt.IfNotExists = $3.(bool)
-		stmt.Options = $6.([]*ast.TableOption)
-		if $7 != nil {
-			stmt.Partition = $7.(*ast.PartitionOptions)
+		stmt := $6.(*ast.CreateTableStmt)
+		stmt.Table = $5.(*ast.TableName)
+		stmt.IfNotExists = $4.(bool)
+		stmt.IsTemporary = $2.(bool)
+		if $2.(bool) {
+			yylex.AppendError(yylex.Errorf("TiDB doesn't support CREATE TEMPORARY TABLE, TEMPORARY will be parsed but ignored."))
+			parser.lastErrorAsWarn()
+		}
+		stmt.Options = $7.([]*ast.TableOption)
+		if $8 != nil {
```

Comment:
why we need to check `$8`?

Reply:
It would be `nil interface{}` if there is no `PartitionOpt`, and the type assertion would panic.